### PR TITLE
IMB enhancement + suggested value support 

### PR DESCRIPTION
**Pull request by rfbgo**
General improvements to IMB application to add better FOMs and bring it inline with what the requesting user wanted 

Example use: 

```
 applications:
    intel-mpi-benchmarks:
      workloads:
        pingpong:
          experiments:
            full_node_{pingpong_type}:
              variables:
                pingpong_type: ['Pingpong', 'Unirandom', 'Multi-Pingpong', 'Birandom', 'Corandom']
                n_ranks: '56'
```

Also adds support for suggested values in the application definition, eg: 

```
    workload_variable('pingpong_type',
                      default='Pingpong',
                      values=['Pingpong', 'Unirandom', 'Multi-Pingpong', 'Birandom', 'Corandom'],
```

This now shows up in `ramble info <app_name>` but does not enforce it

### CODE REVIEW

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/execute_experiment.tpl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#   - workload_run_dir (Will be replaced with the directory of the workload
+#   - application_name (Will be repalced with the name of the application)
+#   - n_nodes (Will be replaced with the required number of nodes)
+#   Any experiment parameters will be available as variables as well.
+
+cd "{experiment_run_dir}"
+
+{spack_setup}
+
+#scontrol show hostnames ${SLURM_JOB_NODELIST} > hostfile
```
Comment (douglasjacobsen) [inline var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/execute_experiment.tpl:22]: I'm not sure why this is in a slurm sub-dir if it doesn't use slurm

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/ramble.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    command: mpirun
+    args:
+    - -n
+    - '{n_ranks}'
+    - -ppn
+    - '{processes_per_node}'
+    - -hostfile
+    - hostfile
+  batch:
+    #submit: sbatch '{execute_experiment}'
```
Comment (douglasjacobsen) [inline var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/ramble.yaml:12]: Slurm comment applies here too.

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/ramble.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  compilers: {}
+  mpi_libraries:
+    impi2018:
+      base: intel-mpi
+      version: 2018.4.274
+      target: x86_64
+  applications:
+    intel-mpi-benchmarks:
+      intel-mpi-benchmarks:
+        base: intel-mpi
```
Comment (douglasjacobsen) [inline var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/ramble.yaml:47]: Intel MPI benchmarks should be installed through a different package than intel-mpi. I think it's called: `intel-mpi-benchmarks` which can then vary independently of the version of MPI we're using for these tests.

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]
A few comments, otherwise this LGTM. Thanks!

File: var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/execute_experiment.tpl (hunk 1 above)
Comment (rfbgo) [reply var/ramble/repos/builtin/applications/intel-mpi-benchmarks/examples/slurm/execute_experiment.tpl]: Great catches all round, address in followup commit

**Comment by rfbgo** [comment]
Thanks for the catches, addressed them in the next commit 👍

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**